### PR TITLE
update tbpore dependencies

### PR DESCRIPTION
**Pull request by mbhall88**
Please do not merge this until I have added the "please review and merge" label (e.g. #34939).

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mbhall88** [comment]
```
package pysam-0.19.0-py27hc2126ed_0 requires libdeflate >=1.10,<1.11.0a0, but none of the providers can be installed
```

I don't understand what is going on with this. None of the dependencies seem to rely on a specific version of libdeflate. And also why is pysam trying to use python 2.7?

@bioconda/build-system can anyone assist with this?

**Comment by leoisl** [comment]
I don't have permissions to push to your `bioconda-recipes` fork :( I am wondering if we fixed `pysam` to `pysam=0.19.0=py38h8bf8b8d_0` (i.e. force to install the build on `py3.8`) if it would work. I have some other suspicions also... it looks like in the last build, [`python 3.10` was installed](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=16830&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=910) but there is [no `pysam` build that supports `python 3.10`](https://anaconda.org/bioconda/pysam/files), thus I guess it defaulted to `python 2.7`? I am wondering if it would work if we just fix the `python` version to `python 3.8`. I can also try these things for you on the other side of the world if I have permissions!

**Comment by mbhall88** [comment]
We shouldn't need pin our python to a single version. It's annoying I can't test locally (due to https://github.com/bioconda/bioconda-utils/issues/772) so have to debug by pushing to this PR. 

Hopefully someone from the bioconda team can help out soon 🤞

**Comment by leoisl** [comment]
This looks ok, all we need to do is downgrade `pysam` to `0.17`? Probably also good to do in `tbpore`, not only in the `conda` recipe, so both have same dependencies versions?

**Comment by mbhall88** [comment]
@BiocondaBot please add label

**Comment by mbhall88** [comment]
@BiocondaBot please update

File: recipes/tbpore/meta.yaml [hunk 1]
```diff
@@ -1,44 +1,43 @@
...
-    - psdm
-    - importlib_metadata >=1.7,<2.0.0
-    - click
-    - cyvcf2
-    - loguru
-    - pyyaml
-    - pandas
-    - pysam
-    - networkx
+    - samtools =1.13
```
Comment (bgruening) [inline recipes/tbpore/meta.yaml:29]: are those strict pins really necessary? Can we loosen them?

Comment (mbhall88) [reply recipes/tbpore/meta.yaml:29]: No, this tool's variant filters were calibrated on a very specific version (1.13) of bcftools unfortunately.

**Comment by mbhall88** [comment]
@BiocondaBot please update

File: recipes/tbpore/meta.yaml (hunk 1 above)
Comment (mbhall88) [reply recipes/tbpore/meta.yaml:29]: @bgruening is that okay?

**Comment by rpetit3** [comment]
@BiocondaBot please fetch artifacts

**Comment by rpetit3** [comment]
@mbhall88 give it a test run and let me know how it turns out

**Comment by mbhall88** [comment]
@BiocondaBot please update

**Comment by mbhall88** [comment]
@rpetit3 works for me

**Review by rpetit3 — APPROVED** [review]

**Comment by rpetit3** [comment]
@BiocondaBot please merge

**Pull request merged**